src/algebra/squares.lean

```diff
@@ -0,0 +1,262 @@
+/-
+Copyright (c) 2021 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import algebra.algebra.basic
+import linear_algebra.finsupp
+import algebra.algebra.tower
+
+/-!
+#  Squares
+
+Let `R` be a Type with `has_mul R`.
+
+We introduce the predicate `is_square` on elements `r ∈ R`, asserting existence of an element
+`s ∈ R`, such that the equality `r = s * s` holds.
+
+Progressively assuming more of the type `R` we
+* introduce the subtype `square R` of squares in `R`;
+* introduce sums of squares, when `R` is an additive commutative monoid;
+* prove an instance of `comm_semigroup` for `squares R`, if `R` is a `comm_semigroup`;
+* prove an instance of `comm_monoid` for `squares R`, if `R` is a `comm_monoid`;
+* prove an instance of `comm_semiring` for `sums_of_square R`, if `R` is a `comm_semiring`.
+-/
+variables {R : Type*}
+
+section is_square
+
+section has_mul
+
+variables (R) [has_mul R]
+
+/--  The predicate `is_square` on `r ∈ R` is the assertion that there exists an element
```

Comment:
IMO using `∈` in code snippets to mean lean's `:` is just confusing.

Reply:
I changed `∈` to `:`.